### PR TITLE
fix: adding validation for aiche registration ID, PDF submission field and getAllUsers query

### PR DESCRIPTION
**Pull request by Srish-ty**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

### Description
- added searching using aiche membership id
- a field for pdf submission in team and individual registration
- Added get all users query for the portal

❤️Thank you!

### Post merge checklist
- [x] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: api/src/apollo/User/mutations.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export const createUser = mutationField('createUser', {
     if (!isMobileValid)
       throw new Error('Invalid Mobile Number, please try again');
 
-    if (args.user.rollNumber) {
+    if (args.user.aicheRegID) {
       const users = await prisma.user.findMany({
         where: {
           rollNumber: args.user.rollNumber,
```
Comment (Shurtu-gal) [inline api/src/apollo/User/mutations.ts:59]: This should also be aiche id then.

File: api/src/apollo/EventRegistration/queries.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const eventRegistration = queryField('eventRegistration', {
   // to fetch list of eventIds for events where user has registered
   description: `Returns a list of events depending upon the arguments`,
   authorize: (_parent, args, ctx) =>
-    args.id || (args.eventID && args.userID)
+    args.orgID
```
[the remark this answers is not in the document]
Comment (Srish-ty) [reply api/src/apollo/EventRegistration/queries.ts:11]: with the previous code, we're not able to fetch eventRegistration info, 
it's giving authorization error

File: api/src/apollo/User/mutations.ts (hunk 1 above)
Comment (Srish-ty) [reply api/src/apollo/User/mutations.ts:59]: It's changed on line no. 63

File: api/src/apollo/Event/mutations.ts [hunk 3]
```diff
@@ -30,16 +30,20 @@ export const createEvent = mutationField('createEvent', {
...
-      [
-        PERMISSIONS.SUPER_ADMIN,
-        PERMISSIONS.SUPER_EDITOR,
-        PERMISSIONS.ORG_ADMIN,
-        PERMISSIONS.ORG_EDITOR,
-      ],
-      args.orgID,
-    ),
+    args.orgID
+      ? checkGqlPermissions(ctx, [])
```
[the remark this answers is not in the document]
Comment (Srish-ty) [reply api/src/apollo/Event/mutations.ts]: changed. https://github.com/dscnitrourkela/project-dates/pull/161/commits/4f27d863837a153512a13f479c798ddf458999a6

File: api/src/apollo/EventRegistration/queries.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Srish-ty) [reply api/src/apollo/EventRegistration/queries.ts]: done. https://github.com/dscnitrourkela/project-dates/pull/161/commits/4f27d863837a153512a13f479c798ddf458999a6

**Comment by Srish-ty** [comment]
@Shurtu-gal, please merge

**Review by Shurtu-gal — APPROVED** [review]

**Pull request merged**